File: src/Java/testapp/src/main/java/com/myservicebus/testapp/DemoScenario.java
package com.myservicebus.testapp;

public final class DemoScenario {
    private DemoScenario() {
    }

    public static String createSubmitMessage(String origin, boolean shouldFault) {
        return "submit:" + (shouldFault ? "fault" : "ok") + ":" + origin;
    }

    public static String createRequestMessage(String origin, boolean shouldFault) {
        return "request:" + (shouldFault ? "fault" : "ok") + ":" + origin;
    }

    public static boolean shouldFaultSubmit(String message) {
        return message != null && message.toLowerCase().startsWith("submit:fault:");
    }

    public static boolean shouldFaultRequest(String message) {
        return message != null && message.toLowerCase().startsWith("request:fault:");
    }
}

File: src/TestApp/DemoScenario.cs
namespace TestApp;

public static class DemoScenario
{
    public static string CreateSubmitMessage(string origin, bool shouldFault) =>
        $"submit:{(shouldFault ? "fault" : "ok")}:{origin}";

    public static string CreateRequestMessage(string origin, bool shouldFault) =>
        $"request:{(shouldFault ? "fault" : "ok")}:{origin}";

    public static bool ShouldFaultSubmit(string? message) =>
        message?.StartsWith("submit:fault:", StringComparison.OrdinalIgnoreCase) == true;

    public static bool ShouldFaultRequest(string? message) =>
        message?.StartsWith("request:fault:", StringComparison.OrdinalIgnoreCase) == true;
}
